dbfit-java/derby/src/main/java/dbfit/environment/DerbyEnvironment.java

```diff
@@ -51,40 +52,193 @@ public Pattern getParameterPattern() {
+                case DatabaseMetaData.procedureColumnInOut:
+                    return Direction.INPUT_OUTPUT;
+                case DatabaseMetaData.procedureColumnOut:
+                    return Direction.OUTPUT;
+                default:
+                    throw new SQLException("Unpexpected procedure parameter type for parameter " +
+                                  rs.getString("COLUMN_NAME"));
+            }
+        }
+    }
+
+    private class DatabaseFunction extends DatabaseObject {
+
+        private DatabaseFunction(DatabaseObjectName objName) {
+            super(objName);
+        }
+
+        @Override
+        public boolean exists() throws SQLException {
+            try (ResultSet rs = getObjects()) {
+                return rs.next();
+            }
+        }
+
+        @Override
+        protected ResultSet getObjects() throws SQLException {
+            return getMetaData().getFunctions(null, name.getSchemaName(), name.getObjectName());
+        }
+
+        @Override
+        public ResultSet getColumns() throws SQLException {
+            return getMetaData().getFunctionColumns(null, name.getSchemaName(), name.getObjectName(), "%");
+        }
+
+        @Override
+        protected Direction getParamDirection(ResultSet rs) throws SQLException {
+            int paramType = rs.getInt("COLUMN_TYPE");
+            switch (paramType) {
+                case DatabaseMetaData.functionColumnIn:
+                    return Direction.INPUT;
+                case DatabaseMetaData.functionReturn:
+                    return Direction.RETURN_VALUE;
+                default:
+                    throw new SQLException("Unpexpected function parameter type for parameter " +
+                                  rs.getString("COLUMN_NAME"));
+            }
+        }
+    }
+
+    private class DatabaseObjectName {
+        String schemaName, objectName;
+
+        private DatabaseObjectName(String defaultSchemaName, String objName) {
+            String[] qualifiers = objName.split("\\.");
+            schemaName = (qualifiers.length == 2) ? qualifiers[0] : defaultSchemaName;
+            objectName = (qualifiers.length == 2) ? qualifiers[1] : qualifiers[0];
+        }
+
+        private String getSchemaName() {
+            return schemaName;
+        }
+
+        String getObjectName() {
+            return objectName;
+        }
+    }
+
+    public Map<String, DbParameterAccessor> getAllProcedureParameters(String callableName)
+                                                                       throws SQLException {
+        String name = callableName.toUpperCase();
+        DatabaseObjectName objName = new DatabaseObjectName(getConnection().getSchema(), name);
+        DatabaseProcedure dbproc = new DatabaseProcedure(objName);
+        if (dbproc.exists()) {
+            return dbproc.getParams();
+        } else {
+            DatabaseFunction dbfunc = new DatabaseFunction(objName);
```

Comment:
Yes, I would usually include the blank lines. Maybe the 1st one could be omitted since the above statement is multi-line with some indentation with 2nd, 3rd lines. But if the `List...` was all on single line - the `for` loop block wouldn't stand out very well without an empty line in between.